### PR TITLE
[JN-984] Fix notifications for trigger page not showing sendgrid status

### PR DESCRIPTION
**Pull request by connorlbark**
#### DESCRIPTION

Fixes bug noticed by Whitney in testing.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

- Go to trigger page
- Open notifications for different triggers
- Ensure that sendgrid statuses are showing up (closed or opened mail icon)

### CODE REVIEW

**Review by MatthewBemis — APPROVED** [review]

File: core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java [hunk 1]
```diff
@@ -56,6 +56,10 @@ public List<Notification> findAllByConfigId(UUID configId, boolean withEnrollees
             attachEnrollees(notifications);
         }
 
+        for (Notification notification : notifications) {
+            dao.attachSendgridEvent(notification);
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java]: this is going to be 1000+ separate DB calls for most production configs.  Let's make a new `attachAllSendgridEvents(List<Notification> notifications)` dao method that grabs all the events in one query

**Review by devonbush — CHANGES_REQUESTED** [review]

File: core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java [hunk 2]
```diff
@@ -257,6 +251,19 @@ public Optional<T> findByProperty(String columnName, Object columnValue) {
         );
     }
 
+    public List<T> findByTriggerId(UUID triggerId) {
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java]: I don't think you need this method (and it doesn't belong in BaseJDbiDao regardless)

File: core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java [hunk 3]
```diff
@@ -56,6 +56,19 @@ public List<Notification> findAllByConfigId(UUID configId, boolean withEnrollees
...
+        //             it shouldn't be too slow since it is only one DB call.
+        sendgridEventDao.findByTriggerId(configId)
+                .forEach(sendgridEvent -> {
+                    Notification notification = notifications
+                            .stream()
+                            .filter(n -> n.getId().equals(sendgridEvent.getNotificationId()))
+                            .findFirst()
+                            .orElseThrow(() -> new RuntimeException("Sendgrid event without notification"));
+                    notification.setEventDetails(sendgridEvent);
+                });
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java:70]:
```suggestion
       List<SendgridEvent> events = sendgridEventDao.findAllByPropertyCollection("notification_id", notifications.stream.map(Notification::getId).toList);
     // now assign them to the right event   
```

**Review by devonbush — CHANGES_REQUESTED** [review]

File: core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java (hunk 2 above)
Comment (connorlbark) [reply core/src/main/java/bio/terra/pearl/core/dao/BaseJdbiDao.java]: Omg I totally fat fingered the Intellij ctrl-click and ended up in the base dao haha

File: core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java (hunk 3 above)
Comment (connorlbark) [reply core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java:70]: Hmm. I am not sure how I feel about creating a SQL query with 1000+ list clause in the where, I assumed that would be undesirable. I'm happy to refactor it to this, though.

File: core/src/test/java/bio/terra/pearl/core/service/notification/NotificationServiceTest.java [hunk 4]
```diff
@@ -73,4 +77,53 @@ void findAllByConfigIdWithEnrollees(TestInfo info) {
...
+        Notification foundNotif2 = notifs.stream().filter(notif -> notif.getId().equals(notif2.getId())).findFirst().orElseThrow();
+        Notification foundNotif3 = notifs.stream().filter(notif -> notif.getId().equals(notif3.getId())).findFirst().orElseThrow();
+        Notification foundNotif4 = notifs.stream().filter(notif -> notif.getId().equals(notif4.getId())).findFirst().orElseThrow();
+
+
+        Assertions.assertEquals(foundNotif1.getEventDetails().getStatus(), "delivered");
+        Assertions.assertNull(foundNotif2.getEventDetails());
+        Assertions.assertEquals(foundNotif3.getEventDetails().getStatus(), "n/a");
+        Assertions.assertEquals(foundNotif4.getEventDetails().getStatus(), "opened");
+    }
```
Comment (devonbush) [inline core/src/test/java/bio/terra/pearl/core/service/notification/NotificationServiceTest.java:132]: nit: check that your intellij is set to always add newlines at the end of files on save

**Review by devonbush — APPROVED** [review]
nice, thanks for the thoughtful discussion

**Pull request merged**